server.go

```diff
@@ -560,18 +560,23 @@ func (srv *Server) serveDNS(m []byte, w *response) {
 	req := new(Msg)
 	req.setHdr(dh)
 
-	switch srv.MsgAcceptFunc(dh) {
+	switch action := srv.MsgAcceptFunc(dh); action {
 	case MsgAccept:
 		if req.unpack(dh, m, off) == nil {
 			break
 		}
 
 		fallthrough
-	case MsgReject:
-		req.SetRcodeFormatError(req)
+	case MsgReject, MsgNotImplemented:
+		if action == MsgNotImplemented {
+			req.SetRcodeNotImplemented(req)
+		} else {
+			req.SetRcodeFormatError(req)
+			req.Zero = false
```

Comment:
only zeroing on formatERR?

also just split the action into 2 cases in the switch

Reply:
The way I structured this code with fallthroughs means we need to have this in one without rewriting it.

That being said, this can just be:
```go
req.SetRcodeFormatError(req)
if action == MsgNotImplemented {
    req.Rcode = RcodeNotImplemented
}
```

Then we can avoid introducing `SetRcodeNotImplemented`.